### ISSUE TITLE
Group similar commands together

### ISSUE THREAD

**Issue by ghost** [backdrop-contrib/b#272]
With the growing list of commands in Bee, it'd be nice to group similar commands together when viewing a list of them all.

For example:
```
 STATE
  state-get             Get the value of a Backdrop state.
  sg, sget

  state-set             Set the value of an existing Backdrop state.
  ss, sset

 USER
  user-login            Display a login link for a given user.
  uli

  user-password         Reset the login password for a given user.
  upw, upwd

  users                 List all user accounts.
  uls, user-list
```

**Comment by ghost** [comment]
Here's a screenshot of `bee help` with the changes from the PR:

![image](https://user-images.githubusercontent.com/2385329/226173475-054a04b4-8c3a-401e-b347-c0c0967f4351.png)

**Comment by yorkshire-pudding** [comment]
Looks a good idea.  I'll try to look in closer detail shortly.  I might need to revisit the groupings in the wiki so they are consistent

**Comment by ghost** [comment]
Here are the groupings I came up with originally:

Config:
- config-export
- config-get
- config-import
- config-set

Core:
- download-core
- install
- update-db

Database:
- db-export
- db-import
- sql

Miscellaneous:
- cache-clear
- cron
- eval
- help
- log
- maintenance-mode
- php-script
- status
- version

Projects:
- disable
- download
- enable
- projects
- theme-admin
- theme-default
- uninstall

State:
- state-get
- state-set

User:
- user-login
- user-password
- users

**Comment by ghost** [comment]
And here are the grouping @yorkshire-pudding is suggesting (I believe they come from the wiki):

Advanced:
- eval
- php-script

Config:
- config-export
- config-get
- config-import
- config-set

Database:
- db-export
- db-import
- sql

Information:
- log
- projects
- status
- version

Installation:
- download-core
- install

Miscellaneous:
- cache-clear
- cron
- help
- maintenance-mode
- state-get
- state-set

Projects:
- disable
- download
- enable
- theme-admin
- theme-default
- uninstall

Update:
- update-db

User:
- user-login
- user-password
- users

**Comment by ghost** [comment]
I'm not opposed to having an 'Advanced' group, in fact I like that it separates the more dangerous/advanced commands from the rest, but I think then I'd prefer to manually move it to the end of the list (I don't want Advanced being the first set of commands people see). I think I'd also like to move `sql` there too, as that's an advanced command that requires knowledge of SQL commands/CLI.

I don't mind the new 'Information' group, and I like that it moves some commands out of 'Miscellaneous', but I'm not sure about putting `projects` there... Yes it's an informative command, but it's also project-specific, so the 'Projects' groups makes sense too. And before you ask, no, I don't want to have commands listed under multiple groups.

I'm not sure about the new 'Installation' or 'Update' groups... I'm not a fan of a group with only one command in it, so perhaps `update-db` should go into 'Database' instead? As for 'Installation' `download-core` technically doesn't have anything to do with installing, so I think 'Core' is still a better name for this group.

I think `state-get` and `state-set` should stay in their own 'State' group.

**Comment by olafgrabienski** [comment]
While we're at it, I'd suggest to rename the "Config" heading to "Configuration".

(If there's need for further discussion, I can open a new issue for it.)

**Comment by yorkshire-pudding** [comment]
> I'm not opposed to having an 'Advanced' group, in fact I like that it separates the more dangerous/advanced commands from the rest, but I think then I'd prefer to manually move it to the end of the list (I don't want Advanced being the first set of commands people see).

Agree that these should be at the bottom; they are in the wiki though I guess it could require a custom exception to do this where you're sorting groups on the fly (some of which could be dynamically added by the API).

> I think I'd also like to move `sql` there too, as that's an advanced command that requires knowledge of SQL commands/CLI.

Yes, it is advanced but it is a very natural fit within the database group.  I could go either way on this, so if you feel strongly about this, then I'm fine with that.

> I don't mind the new 'Information' group, and I like that it moves some commands out of 'Miscellaneous', but I'm not sure about putting `projects` there... Yes it's an informative command, but it's also project-specific, so the 'Projects' groups makes sense too. And before you ask, no, I don't want to have commands listed under multiple groups.

I could go either way on this. In the wiki I called it "Project Actions" where it clearly doesn't belong.  I think that either:
- call the category "Project Actions" and then it definitely belongs in "Information"
- rename in Wiki as "Projects" and then it can go in there.

> I'm not sure about the new 'Installation' or 'Update' groups... I'm not a fan of a group with only one command in it, so perhaps `update-db` should go into 'Database' instead? 

There is a work in progress on #111 that could see an `update` command in the near future; in which case, it would not be on its own.

`update-db` is a bit of a misnomer as it applies to any update hooks which may or may not affect the database (the majority of contrib ones I've dealt with actually don't touch the database, but rather config).  Also, it will apply update hooks from contrib so it definitely doesn't belong in "Core"

> As for 'Installation' `download-core` technically doesn't have anything to do with installing, so I think 'Core' is still a better name for this group.

Really? Please tell me more.  That is the only thing I have used it for. 

> I think `state-get` and `state-set` should stay in their own 'State' group.

Ok

> While we're at it, I'd suggest to rename the "Config" heading to "Configuration".

That makes sense. Thanks for the suggestion @olafgrabienski  . Doesn't need a separate issue

**Comment by ghost** [comment]
>> I think I'd also like to move sql there too, as that's an advanced command that requires knowledge of SQL commands/CLI.
> 
> Yes, it is advanced but it is a very natural fit within the database group. I could go either way on this, so if you feel strongly about this, then I'm fine with that.

Yeah, let's move it. If people are looking for this I'd rather they find it under 'Advanced'. Kinda tells them to tread lightly.

> I think that either:
> - call the category "Project Actions" and then it definitely belongs in "Information"
> - rename in Wiki as "Projects" and then it can go in there.

I'm not a fan of 'Project Actions', so let's do the latter. Also, `users` is an informative command, but it still fits best under 'User'. I think we can define 'Information' as a group for informative commands that don't fit better elsewhere. Let's also move `help` under 'Information'. We can update the wiki once the PR is merged to make sure everything matches up.

> There is a work in progress on https://github.com/backdrop-contrib/bee/issues/111 that could see an update command in the near future; in which case, it would not be on its own.

Ok, let's keep the 'Update' group then. It's a shame about the name of the `update-db` command. Can't think of better name for that though...

>> As for 'Installation' download-core technically doesn't have anything to do with installing, so I think 'Core' is still a better name for this group.
>
> Really? Please tell me more. That is the only thing I have used it for.

`download-core` just downloads a copy of Backdrop into the current/specified directly. Sure, you'll usually then install it with a database, but that's a separate command and there are a (very) few use-cases where you might not do that. That's why I used the term 'technically'.

'Core' covers both downloading core and installing core, so I think we should name the group for these two commands that instead.

> That makes sense. Thanks for the suggestion @olafgrabienski . Doesn't need a separate issue

Done.

I've also renamed 'User' to 'Users' to match the plural form of 'Projects'.

Finally, I could be convinced to move `theme-admin` and `theme-default` into a new 'Themes' group. I know they're still projects, but most of the project commands apply to all project types whereas these are only for themes. But I understand if others disagree, especially since `uninstall` only applies to modules and that still lives under the 'Projects' group.

So here's what we now have:

Configuration:
- config-export
- config-get
- config-import
- config-set

Core:
- download-core
- install

Database:
- db-export
- db-import

Information:
- help
- log
- status
- version

Miscellaneous:
- cache-clear
- cron
- maintenance-mode

Projects:
- disable
- download
- enable
- projects
- theme-admin
- theme-default
- uninstall

State:
- state-get
- state-set

Update:
- update-db

Users:
- user-login
- user-password
- users

Advanced:
- eval
- php-script
- sql

**Comment by yorkshire-pudding** [comment]
@BWPanda - I'm happy with those explanations.

I could be persuaded to have a separate theme category; it could make sense (though not urgent) to actually put the theme debug setting as a commands rather than people having to know the `config-set` command. This are exposed by `devel` but otherwise this core setting is unavailable.

**Comment by ghost** [comment]
Awesome, here's an updated (and hopefully final) list with the new 'Themes' group:

Configuration:
- config-export
- config-get
- config-import
- config-set

Core:
- download-core
- install

Database:
- db-export
- db-import

Information:
- help
- log
- status
- version

Miscellaneous:
- cache-clear
- cron
- maintenance-mode

Projects:
- disable
- download
- enable
- projects
- uninstall

Themes:
- theme-admin
- theme-default

State:
- state-get
- state-set

Update:
- update-db

Users:
- user-login
- user-password
- users

Advanced:
- eval
- php-script
- sql

> it could make sense (though not urgent) to actually put the theme debug setting as a commands rather than people having to know the config-set command. This are exposed by devel but otherwise this core setting is unavailable.

@yorkshire-pudding Want to create a separate issue for that so it's not forgotten?

**Comment by yorkshire-pudding** [comment]
All looks good.

> > it could make sense (though not urgent) to actually put the theme debug setting as a commands rather than people having to know the config-set command. This are exposed by devel but otherwise this core setting is unavailable.
> 
> @yorkshire-pudding Want to create a separate issue for that so it's not forgotten?

Done  @ #277

**Comment by ghost** [comment]
PR updated.
